backend/src/sdk.cpp

```diff
@@ -209,6 +240,147 @@ restinio::request_handling_status_t SDK::handleTxSDKCall(const restinio::request
     return req->create_response().set_body(absl::StrJoin(outData, ",")).done();
 }
 
+void SDK::handleSetStationState(const nlohmann::json json)
+{
+    if (!mClient->IsVoiceConnected()) {
+        TRACK_LOG_ERROR("kSetStationState requires a voice connection");
+        return;
+    }
+
+    if (!json["value"].contains("frequency")) {
+        TRACK_LOG_ERROR("kSetStationState requires a frequency");
+        return;
+    }
+
+    int frequency = json["value"]["frequency"];
+
+    mClient->SetRadioGainAll(UserSession::currentRadioGain);
+
+    if (json["value"].contains("rx")) {
+        bool oldRxValue = mClient->GetRxState(frequency);
+        bool newRxValue;
+
+        // rx is either a "toggle" string or a boolean value
+        if (json["value"]["rx"].is_string() && json["value"]["rx"] == "toggle") {
+            newRxValue = !oldRxValue;
+
+            TRACK_LOG_INFO("Toggling rx for {}, new value {}", frequency, newRxValue);
+            mClient->SetRx(frequency, newRxValue);
+        } else {
+            newRxValue = json["value"]["rx"].get<bool>();
+
+            TRACK_LOG_INFO("Setting rx for {} to {}", frequency, newRxValue);
+            mClient->SetRx(frequency, newRxValue);
+        }
+
+        if (!oldRxValue && newRxValue) {
+            // When turning on RX, we refresh the transceivers
+            auto states = mClient->getRadioState();
+            if (states.find(frequency) != states.end() && !states[frequency].stationName.empty()) {
+                mClient->FetchTransceiverInfo(states[frequency].stationName);
+            }
+        }
```

Comment:
This bugs me. It's a direct copy/paste of the code from main.cpp when rx is set via the UI. It really feels like there should be some code consolidation so the UI and websocket just call the same method somewhere to do this stuff.